klippy/extras/rpi_temperature.py

```diff
@@ -0,0 +1,89 @@
+# Support for Raspberry Pi temperature sensor
+#
+# Copyright (C) 2020  Al Crate <al3ph@users.noreply.github.com>
+#
+# This file may be distributed under the terms of the GNU GPLv3 license.
+
+import logging
+
+RPI_REPORT_TIME = 1.0
+KELVIN_TO_CELSIUS = -273.15
+
+class RPiTemperature:
+    def __init__(self, config):
+        self.printer = config.get_printer()
+        self.name = config.get_name().split()[-1]
+        self.reactor = self.printer.get_reactor()
+
+        self.temp = 0.0
+        self.file_handle = None
+        self.min_temp = config.getfloat('min_temp', -40,
+                                        minval=KELVIN_TO_CELSIUS)
+        self.max_temp = config.getfloat('max_temp', 85,
+                                        above=self.min_temp)
```

Comment:
The code shouldn't read the min_temp/max_temp directly from the config.  Instead, it should obtain the range from the `setup_minmax()` method.

Reply:
How do I define default values using that ?

Reply:
You can't default a min/max value in the sensor.  Only the owner of the sensor can.  For example, see how `klippy/extras/temperature_sensor.py` sets a default.

-Kevin

Reply:
I've no idea how to do that, I tried but I get a recursion error, there's just not enough information in the comments/ docs for me to know what I'm doing.